redux-form/lib/Field.d.ts

```diff
@@ -91,13 +92,13 @@ interface BaseFieldProps {
  * @param previousAllValues All the values in the entire form before the current change.
  *                          This will be an Immutable Map if you are using Immutable JS.
  */
-export type Normalizer = (value: FieldValue, previousValue?: FieldValue, allValues?: Object, previousAllValues?: Object) => FieldValue;
+export type Normalizer = (value: FieldValue, previousValue?: FieldValue, allValues?: any, previousAllValues?: any) => FieldValue;
 
 export type Formatter = (value: FieldValue, name: string) => FieldValue;
 
 export type Parser = (value: FieldValue, name: string) => FieldValue;
 
-export type Validator = (value: FieldValue, allValues?: Object) => undefined | string;
+export type Validator = (value: FieldValue, allValues?: any, props?: any) => any;
```

Comment:
Well, this would be kind of BC, as some other wrapping types have to become generics as well. 

I think we need to go through the definitions sometime as in some places generics are not really needed or could be simplified.